query-engine/query-engine/src/server.rs

```diff
@@ -156,6 +183,28 @@ impl HttpServer {
         Ok(res)
     }
 
+    fn handle_debug_headers(headers: &HashMap<String, String>) -> Option<Response<Body>> {
+        match (headers.get(DEBUG_FATAL_HEADER), headers.get(DEBUG_NON_FATAL_HEADER)) {
+            (Some(_fatal), _) => {
+                info!("Query engine debug fatal error, shutting down.");
+                std::process::exit(1)
+            }
+            (_, Some(_nonfatal)) => {
+                let err = user_facing_errors::Error::from_panic_payload(&String::from("Debug panic"));
+                let bytes = serde_json::to_vec(&err).unwrap();
```

Comment:
I don't have the full context on the feature request, so I don't know if it should be a valid GraphQL response or not. I think the error will only serialize to a small object that isn't a complete GraphQL response.

Reply:
Good point, will ask again.

Reply:
For now this is alright, if different requirements arise we'll revisit.